### PR TITLE
Ksnelson/vz 6749

### PR DESCRIPTION
**Pull request by knelson7**
two new methods to the ComponentValidator interface and added default implementation for each component in registry

### CODE REVIEW

File: platform-operator/controllers/verrazzano/component/spi/component.go [hunk 1]
```diff
@@ -109,6 +109,10 @@ type ComponentValidator interface {
 	ValidateInstall(vz *vzapi.Verrazzano) error
 	// ValidateUpdate checks if the specified new Verrazzano CR is valid for this component to be updated
 	ValidateUpdate(old *vzapi.Verrazzano, new *vzapi.Verrazzano) error
+	// ValidateInstall checks if the specified Verrazzano CR is valid for this component to be installed
+	ValidateInstallV1Beta1(vz *vzapi.Verrazzano) error
```
Comment (anders-swanson) [inline platform-operator/controllers/verrazzano/component/spi/component.go:113]: This needs to be the v1beta1.Verrazzano package. vzapi is an import alias for v1alpha1.Verrazzano

Comment (anders-swanson) [reply platform-operator/controllers/verrazzano/component/spi/component.go]: The type of all the parameters for the v1beta1 validate methods must be a v1beta1.Verrazzano. The v1beta1 validating webhook will be passing down objects of type v1beta1.Verrazzano

**Review by anders-swanson — APPROVED** [review]
LGTM

File: platform-operator/controllers/verrazzano/component/appoper/app_operator_component.go [hunk 2]
```diff
@@ -127,6 +127,11 @@ func (c applicationOperatorComponent) ValidateUpdate(old *vzapi.Verrazzano, new
 	return c.HelmComponent.ValidateUpdate(old, new)
 }
 
+// ValidateUpdate checks if the specified new Verrazzano CR is valid for this component to be updated
+func (c applicationOperatorComponent) ValidateUpdateV1Beta1(old *installv1beta1.Verrazzano, new *installv1beta1.Verrazzano) error {
```
Comment (michael-cico) [inline platform-operator/controllers/verrazzano/component/appoper/app_operator_component.go:131]: Is this needed, or will it pick up the `HelmComponent` version by default?

Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/appoper/app_operator_component.go:131]: Same for the other components as well.

File: platform-operator/controllers/verrazzano/component/spi/component.go (hunk 1 above)
Comment (michael-cico) [reply platform-operator/controllers/verrazzano/component/spi/component.go]: should the version be an argument and avoid version-specific interface methods, or will it be more complicated than its worth for the number of API versions we may be supporting at any given time (down the road I mean).

File: platform-operator/controllers/verrazzano/component/appoper/app_operator_component.go (hunk 2 above)
Comment (anders-swanson) [reply platform-operator/controllers/verrazzano/component/appoper/app_operator_component.go:131]: These stubs are going to get filled in as we implement component-level validators. It's not strictly necessary at this point but also will not hurt anything

**Review by michael-cico — APPROVED** [review]

**Review by anders-swanson — APPROVED** [review]
LGTM

**Pull request merged**